src/always-be-playing.js

```diff
@@ -135,6 +121,71 @@ export default class GapSkipper {
   }
 
   /**
+   * Handler for `waiting` events from the player
+   *
+   * @private
+   */
+  waiting_() {
```

Comment:
There is a lot going on in this function. I think it would be helpful for readability if the different checks ("check to see if we fell out of the live window", "check for video underflow", "check for gap") were made out into their own functions